django_project/monitor/observation_views.py

```diff
@@ -11,6 +11,118 @@
+from django.views.decorators.csrf import csrf_exempt
+from django.contrib.auth.decorators import login_required
+import json
+from django.contrib.gis.geos import Point
+from decimal import Decimal
+
+@csrf_exempt
+@login_required
+def create_observations(request):
+    if request.method == 'POST':
+        try:
+            # Parse JSON data from the request body
+            data = json.loads(request.body.decode('utf-8'))
+
+            # Extract datainput from the payload
+            datainput = data.get('datainput', {})
+
+            # Extract other fields from the payload
+            flatworms = data.get('flatworms', False)
+            leeches = data.get('leeches', False)
+            crabs_shrimps = data.get('crabs_shrimps', False)
+            stoneflies = data.get('stoneflies', False)
+            minnow_mayflies = data.get('minnow_mayflies', False)
+            other_mayflies = data.get('other_mayflies', False)
+            damselflies = data.get('damselflies', False)
+            dragonflies = data.get('dragonflies', False)
+            bugs_beetles = data.get('bugs_beetles', False)
+            caddisflies = data.get('caddisflies', False)
+            true_flies = data.get('true_flies', False)
+            snails = data.get('snails', False)
+            score = Decimal(str(datainput.get('score', 0)))
+            comment = datainput.get('notes', '')
+            water_clarity = Decimal(str(datainput.get('waterclaritycm', 0)))
+            water_temp = Decimal(str(datainput.get('watertemperatureOne', 0)))
+            ph = Decimal(str(datainput.get('ph', 0)))
+            diss_oxygen = Decimal(str(datainput.get('dissolvedoxygenOne', 0)))
+            diss_oxygen_unit = datainput.get('dissolvedoxygenOneUnit', 'mgl')
+            elec_cond = Decimal(str(datainput.get('electricalconduOne', 0)))
+            elec_cond_unit = datainput.get('electricalconduOneUnit', 'mS/m')
+            site_id = datainput.get('selectedSite')
+            try:
+                site_id = int(site_id)
+            except (ValueError, TypeError):
+                site_id = 0
+            obs_date = datainput.get('date')
+
+            # Get the user from the request object
+            user = request.user
+
+            # Check if the site with the given site_id exists
+            try:
+                site = Sites.objects.get(gid=site_id)
+            except Sites.DoesNotExist:
+                # If it doesn't exist, create a new site
+                site_name = datainput.get('siteName', '')
+                river_name = datainput.get('riverName', '')
+                description = datainput.get('siteDescription', '')
+                river_cat = datainput.get('rivercategory', '')
+                # longitude = Decimal(datainput.get('longitude',0))
+                # latitude = Decimal(datainput.get('latitude',0))
+                # TODO still to merge with Irwan for saving location
+                longitude = Decimal('30.12345')
```

Comment:
Please use datainput.get('longitude', 0)

Reply:
@meomancer also if you check in network tab response ...it returns this when i trying adding the the_geom point field {
    "status": "error",
    "message": "Invalid parameters given for Point initialization."
} thats why I commented it out for now